### PR TITLE
Security and Surveillance pull request

### PR DESCRIPTION
**Pull request by pradeepts**
Hi Team,

The pull request contains the following changes which are added to the core solution

1. Modularized templates for Basic, Standard and Premium 
2. Build code with Application Insights and OMS 
3. Templates with HA components
4. Added documentation(readme.md files ) for Updated solution

Please review the same and let us know if you need any changes.

Thanks,

### CODE REVIEW

File: documentation/Deployment-Guide.md [hunk 1]
```diff
@@ -0,0 +1,1136 @@
...
+| **OMS Location**   | Choose location for OMS Log Analytics to deploy. For more information,see https://docs.microsoft.com/en-us/azure/log-analytics/log-analytics-overview   | australiasoutheast, canadacentral, centralindia, eastus, japaneast, southeastasia, uksouth, westeurope  | eastus  |
+| **App Insights Location**  | specify the region for application insights   | eastus, northeurope, southcentralus, southeastasia, westeurope, westus2 | westus2 |
+| **Location DR**  | Choose locationDR which is different from the resource group region,if you have selected standard (or) premium option in the solution type inputparameter, for more information about the locationDr navigate to https://github.com/sysgain/SecurityAndSurveillance/blob/master/documentation/Deployment-Guide.md   | NA, EastUS2, CentralUS, WestUS2, WestCentralUS, CanadaCentral, CanadaEast, AustraliaEast, AustraliaSouthEast, CentralIndia,  SouthIndia, EastAsia, SouthEastAsia, JapanWest, JapanEast, KoreaCentral,    KoreaSouth, UKSouth, UKWest | NA |
+| **Traffic Manager Name**    | Please provide the name of the trafficmanager certficate custome domain name   |    | NA |
```
Comment (githubmobiliya) [inline documentation/Deployment-Guide.md]: Typo "Custome" should be "custom"

File: documentation/Deployment-Guide.md [hunk 2]
```diff
@@ -0,0 +1,1136 @@
...
+•	Apply tags to resources to logically organize all the resources in your subscription.
+
+
+### 5.1 ARM Template Deployment Using Azure Portal
+
+1.	Click the below **Git hub** repo URL.
+
+https://github.com/sysgain/SecurityAndSurveillance/tree/master
+
+2. Select **main-template.json** from **master** branch, click on **Raw** from the top right corner.
```
Comment (githubmobiliya) [inline documentation/Deployment-Guide.md]: **main-template.json** --> not found in the repo.

**Comment by githubmobiliya** [comment]
Please review the pull request for similar comments as above.

File: documentation/Deployment-Guide.md (hunk 1 above)
Comment (pradeepts) [reply documentation/Deployment-Guide.md]: We have updated the above comment, please review and close it.

File: documentation/Deployment-Guide.md (hunk 2 above)
Comment (pradeepts) [reply documentation/Deployment-Guide.md]: we have updated the correct file name i.e. maintemplate.json . Please review and close it.

**Comment by pradeepts** [comment]
We have reviewed the comments and updated accordingly, kindly review the same.

File: documentation/Deployment-Guide.md [hunk 3]
```diff
@@ -0,0 +1,1137 @@
+# Microsoft
+
+# Securiy Surveillance
```
Comment (githubmobiliya) [inline documentation/Deployment-Guide.md]: Spell Correction Required: Security

Comment (pradeepts) [reply documentation/Deployment-Guide.md]: Updated the document with correct spelling.

**Comment by githubmobiliya** [comment]
> Hi Team,
> 
> The pull request contains the following changes which are added to the core solution
> 
> 1. Modularized templates for Basic, Standard and Premium
> 2. Build code with Application Insights and OMS
> 3. Templates with HA components
> 4. Added documentation(readme.md files ) for Updated solution
> 
> Please review the same and let us know if you need any changes.
> 
> Thanks,

Hi Pradeep,

We tested deployment for basic plan and there were deployment errors.
Can you review and check the issues for the same. 

Attached maintemplate and parameters files for reference.

Error1 (1)

Error1 (2)

Error1 (3)
[maintemplate.parameters.zip](https://github.com/MobiliyaTechnologies/SecurityAndSurveillance/files/3078810/maintemplate.parameters.zip)

**Comment by pradeepts** [comment]
Hi Team,

As per the updates to the Azure, we have removed E2EDiagnostics, OMS for web application and added the following properties to the IoTHub resource.
{

                                "category": "DistributedTracing",

                                "enabled": true,

                                "retentionPolicy": {

                                    "days": 0,

                                    "enabled": false

                                }  

                            },

                            {

                                "category": "Configurations",

                                "enabled": true,

                                "retentionPolicy": {

                                    "days": 0,

                                    "enabled": false

                                }  

                            },

                            {

                                "category": "DeviceStreams",

                                "enabled": true,

                                "retentionPolicy": {

                                    "days": 0,

                                    "enabled": false

                                }  

                            }
After updating the above details we were able to deploy the template successfully. PFB Screenshot.
 
MicrosoftTeams-image (24)
MicrosoftTeams-image (25)

The updated nested templates can be found in the same pull request. Please deploy the main-template.json and let us know if you need anything from us
_NOTE_: There is no change in the main-template.json file. The changes are made in the following nested templates(files) 
iothub.json
webapp.json
webappha.json
webapp-dr.json

**Pull request merged**